### ISSUE TITLE
January 2021 Release

### ISSUE THREAD

**Issue by toasted-nutbread** [FooSoft/yomichan#1340]
https://github.com/FooSoft/yomichan/releases/tag/21.1.31.0

@FooSoft Below is a general checklist for various steps in the release process. Feel free to check off the items or change it as you feel necessary, and if you think that works well, we can just copy-paste that for future releases.

Note that for the Firefox auto-updating testing release, you may want to just link to the version you would upload on the Github release, since the [current download link](https://foosoft.net/projects/yomichan/dl/yomichan_testing.xpi) isn't versioned. Or you could upload versioned files there as well, whatever works for you.

### General
* [x] Create [release tag](https://github.com/FooSoft/yomichan/tree/21.1.31.0).
* [x] Create [GitHub release](https://github.com/FooSoft/yomichan/releases/tag/21.1.31.0).

### Testing
* [x] Update Chrome testing release.
* [x] Update Firefox testing release.
* [x] Update Firefox [updates.json](https://github.com/FooSoft/foosoft.github.io/blob/master/projects/yomichan/dl/updates.json) and [verify](https://foosoft.net/projects/yomichan/dl/updates.json).
* [x] Upload testing builds to the [GitHub release](https://github.com/FooSoft/yomichan/releases/tag/21.1.31.0).

### Stable
* [x] Update Chrome stable release.
* [x] Update Firefox stable release.
* [x] Upload stable builds to the [GitHub release](https://github.com/FooSoft/yomichan/releases/tag/21.1.31.0).

**Comment by FooSoft** [comment]
Well that's annoying, apparently you can't attach XPI files to releases... this means that we probably will not be able to host the testing packages on GitHub (for auto updates at least). I'll switch my site to just keep old versions around and have the existing link point to the latest one in the series.

**Comment by FooSoft** [comment]
Never mind, got it to work. it appears it wasn't working properly because I was uploading multiple files at once? Who knows. Anyway, for the time being I'll host the testing releases on my website and if that introduces scaling issues (too many binary blobs in git), I'll just make the links point to the downloads on the release page 👍🏻

**Comment by toasted-nutbread** [comment]
If binary blobs are an issue, you can set up git LFS, which doesn't affect future workflows at all beyond the initial setup commit.

**Comment by FooSoft** [comment]
From my experience, LFS has always been kind of jankey on GitHub, and you have actually more quotas to consider with that than normal git. I'm not too concerned about this to be honest, and versioning releases has obvious benefits.

**Comment by toasted-nutbread** [comment]
So far, the only issues I've noticed have been #1346 and #1376, which are pretty marginal. IMO not worth an immediate patch unless something else comes up, since we're already halfway through the month.

**Comment by FooSoft** [comment]
Sounds good, I'll ship the mainline release 👍🏻

**Comment by Thermospore** [comment]
Hey guys, I'm liking the new update a lot. Here are some issues I've found:

First off my settings file and browser version:
[yomichan-settings-2021-02-17-01-12-44.zip](https://github.com/FooSoft/yomichan/files/5992570/yomichan-settings-2021-02-17-01-12-44.zip)
>Google Chrome は最新版です
>バージョン: 88.0.4324.182（Official Build） （64 ビット）

1. Sometimes entries have various elements of their frequency tags in the wrong order

![more scrambled stuff](https://user-images.githubusercontent.com/35350819/108143014-571cfe80-7073-11eb-8949-495e6c8a46d2.png)

![stuff reversed](https://user-images.githubusercontent.com/35350819/108142719-c7775000-7072-11eb-9fbf-1fe7d0f179de.png)

![more reversed](https://user-images.githubusercontent.com/35350819/108142910-2b9a1400-7073-11eb-87d7-7f0234ae31d5.png)

Now that I'm writing this, it seems like the order is based on which form you scanned. I think it's a lot more readable to consistently have the `(_ only)` tags in the same order as the top of the definition, and the frequency dictionaries to always be in the same order you specified for them

2. Speaking of which, this isn't new, but the priority number for frequency dictionaries has no effect

(pic for reference)
![image](https://user-images.githubusercontent.com/35350819/108143703-b4fe1600-7074-11eb-97b3-7a0005a3fde6.png)

The workaround for this was to remove them all, then re-import them in the order you wanted them to be displayed

3. ctrl c and ctrl f no longer work in the pop up menu, but they still work on the search page. Looks like this one might already be addressed tho #1411

(pic for reference)
![ctrl C or F doesnt work A - works in search tho](https://user-images.githubusercontent.com/35350819/108143990-335ab800-7075-11eb-8b26-3212688d5d3a.png)

I tried setting a different keyboard shortcut for copy, but it still doesn't seem to work
![ctrl C or F doesnt work B - works in search tho](https://user-images.githubusercontent.com/35350819/108144002-381f6c00-7075-11eb-9e8f-151fd1b83c4a.png)

For now I'm just using the context menu to copy

4. I think for compact mode it would make sense to also omit the `(` and the `only)` in the frequency tags. It is supposed to be compact after all :^) they tend to pile up quick

5. Something funky is going on with the audio; it will randomly replay sometimes even when the window is out of focus. That's all I can say for now, but I'll keep an eye out

Thanks for taking a look!
-Tyler

**Comment by toasted-nutbread** [comment]
1. The implementation doesn't attempt to sort them in this order, but this can be done. Can you create a new issue for that for tracking?
2. I'll have to look into this some more, this worked at some point, although how it worked was a bit strange depending on the display mode.
3. Indeed, that's fixed.
4. Custom CSS can suppress those. Not all users may have many frequency dictionaries installed, so it may be acceptable to some users.
5. Let me know if you discover something. The audio system for results was refactored a bit, but nothing should randomly play unless auto-play is enabled. When you say "window", what do you mean? The search page, or the embedded popup?

**Comment by toasted-nutbread** [comment]
# Release 21.1.31.1
https://github.com/FooSoft/yomichan/releases/tag/21.1.31.1
@FooSoft This fixes some bugs and addresses some of the user feedback. I will continue to address the feedback for the display of the frequency tags, but I feel that is less urgent.

### General
* [x] Create [release tag](https://github.com/FooSoft/yomichan/tree/21.1.31.1).
* [x] Create [GitHub release](https://github.com/FooSoft/yomichan/releases/tag/21.1.31.1).

### Testing
* [ ] Update Chrome testing release.
* [ ] Update Firefox testing release.
* [ ] Update Firefox [updates.json](https://github.com/FooSoft/foosoft.github.io/blob/master/projects/yomichan/dl/updates.json) and [verify](https://foosoft.net/projects/yomichan/dl/updates.json).
* [ ] Upload testing builds to the [GitHub release](https://github.com/FooSoft/yomichan/releases/tag/21.1.31.1).

### Stable
* [ ] Update Chrome stable release.
* [ ] Update Firefox stable release.
* [ ] Upload stable builds to the [GitHub release](https://github.com/FooSoft/yomichan/releases/tag/21.1.31.1).

**Comment by Thermospore** [comment]
Wow, the patch is already up, what a legend

Thanks for the response @toasted-nutbread 
1. Sure thing
2. Sounds good, thanks
3. Thanks, yea both ctrl c and ctrl f are working for me in 21.1.31.1!
4. Fair enough. eyyyyy I got it to work!
```
.frequency-disambiguation:before{ content: ""; }
.frequency-disambiguation:after{ content: ""; margin-right: 1px; }
```
Without:
![image](https://user-images.githubusercontent.com/35350819/108113047-6dab6180-7043-11eb-8a75-dfbc80e1b1ca.png)

With:
![image](https://user-images.githubusercontent.com/35350819/108160724-197d9d00-7096-11eb-8ee2-dada9161d3e8.png)

Credit to @shoui520, I modified the CSS from [their site](https://learnjapanese.moe/yomicss/#frequency-value-outside-of-tag). Maybe I would know more CSS if I spent less time looking at the settings page 🙂 (inside joke)

5. IIRC when this was happening I had one chrome tab with an embedded popup opened, another tab with the search page (might have had the same lookup as the embedded popup), and the opened tab was unrelated to yomichan. The last thing I did in one of the yomichan instances was play the entry's audio. While I was doing stuff outside of chrome, occasionally one of the yomichan instances replayed the word a couple times in a row. Might have been triggered by a keyboard action, not sure. yomichan hotkeys are default and Auto-play is disabled. I'll see if it happens again tonight

**Comment by FooSoft** [comment]
Yup @toasted-nutbread is amazing for jumping on the odd issues that were not caught in testing 💯

**Comment by byjtje** [comment]
The `Use this new settings page by default` option does nothing after the update, it always defaults to the new settings page

Though I guess it doesn't matter if the old settings are planned to get phased out

**Comment by Thermospore** [comment]
Messing around with the new regex feature. Might be user error, but there seems to be some funkyness around scanning half width latin characters? Was easiest to just show it in a quick vid: https://youtu.be/j7f4QirKwWM

**Comment by toasted-nutbread** [comment]
> 
> 
> The `Use this new settings page by default` option does nothing after the update, it always defaults to the new settings page
> 
> Though I guess it doesn't matter if the old settings are planned to get phased out

Thought I fixed it in the patch, but I guess not. #1421

**Comment by toasted-nutbread** [comment]
> Messing around with the new regex feature. Might be user error, but there seems to be some funkyness around scanning half width latin characters? Was easiest to just show it in a quick vid: https://youtu.be/j7f4QirKwWM

Fixed in #1423
